src/Oro/Component/EntitySerializer/EntitySerializer.php

```diff
@@ -275,36 +289,53 @@ protected function serializeItem($entity, $entityClass, $config)
         $result         = [];
         $entityMetadata = $this->doctrineHelper->getEntityMetadata($entityClass);
         $resultFields   = $this->getFieldsToSerialize($entityClass, $config);
+
         foreach ($resultFields as $field) {
-            $value = null;
-            if ($this->dataAccessor->tryGetValue($entity, $field, $value)) {
-                $targetConfig = $this->getFieldConfig($config, $field);
-                if ($entityMetadata->isAssociation($field)) {
-                    if ($value !== null) {
-                        if (!empty($targetConfig['fields'])) {
-                            if (is_string($targetConfig['fields'])) {
-                                $value = $this->dataAccessor->getValue($value, $targetConfig['fields']);
-                                $value = $this->dataTransformer->transform(
-                                    $entityClass,
-                                    $field,
-                                    $value,
-                                    $targetConfig
-                                );
-                            } else {
-                                $value = $this->serializeItem(
-                                    $value,
-                                    $entityMetadata->getAssociationTargetClass($field),
-                                    $targetConfig
-                                );
-                            }
-                        } else {
-                            $value = $this->dataTransformer->transform(
-                                $entityClass,
-                                $field,
-                                $value,
-                                $targetConfig
-                            );
-                        }
+            if ($this->authChecker && !$this->authChecker->isGranted('VIEW', new FieldVote($entity, $field))) {
+                continue;
+            }
```

Comment:
Guys, let me know if you don't like the fact that EntitySerializer component knows about SecurityComponent's FieldVote and AuthorizationChecker. Other option might be to override component's methods in currently deprecated \Oro\Bundle\SoapBundle\Serializer\EntitySerializer.

cc @yurio @vsoroka